Tests/SourceKitLSPTests/LocalSwiftTests.swift

```diff
@@ -1476,4 +1476,55 @@ final class LocalSwiftTests: XCTestCase {
     data = EditorPlaceholder(text)
     XCTAssertNil(data)
   }
+  
+  func testIncrementalParse() throws {
+    let url = URL(fileURLWithPath: "/\(UUID())/a.swift")
+    let uri = DocumentURI(url)
+
+    sk.allowUnexpectedNotification = false
+    
+    sk.send(DidOpenTextDocumentNotification(textDocument: TextDocumentItem(
+      uri: uri,
+      language: .swift,
+      version: 0,
+      text: """
+      func foo() {
+      }
+      class bar {
+      }
+      """
+    )))
+    
+    let didChangeTextDocumentExpectation = XCTestExpectation(description: "didChangeTextDocument")
+    sk.sendNoteSync(DidChangeTextDocumentNotification(textDocument: .init(uri, version: 1), contentChanges: [
+      .init(range: Range(Position(line: 2, utf16index: 7)), text: "a"),
+    ]), { (note: LanguageServerProtocol.Notification<PublishDiagnosticsNotification>) -> Void in
+      log("Received diagnostics for text edit - syntactic")
+      didChangeTextDocumentExpectation.fulfill()
+    }, { (note: LanguageServerProtocol.Notification<PublishDiagnosticsNotification>) -> Void in
+      log("Received diagnostics for text edit - semantic")
+    })
+    
+    let result = XCTWaiter.wait(for: [didChangeTextDocumentExpectation], timeout: defaultTimeout)
```

Comment:
And if you’re using `self.expectation`, you should also be using `self.wait`. IIRC it will fail the test if the expectation isn’t fulfilled, so you don’t need to check `result` either.

```suggestion
    let result = self.wait(for: [didChangeTextDocumentExpectation], timeout: defaultTimeout)
```